tests/integration/test_movielens.py

```diff
@@ -68,49 +67,51 @@ def test_load_pandas_df(
     title_example,
     genres_example,
     year_example,
+    tmpdir,
 ):
     """Test MovieLens dataset load into pd.DataFrame
     """
     # Test if correct data are loaded and local_cache_path works
-    with TemporaryDirectory() as tmp_dir:
-        # Test if can handle different size of header columns
-        header = ["a"]
-        df = load_pandas_df(size=size, local_cache_path=tmp_dir, header=header)
-        assert len(df) == num_samples
-        assert len(df.columns) == max(
-            len(header), 2
-        )  # Should load at least 2 columns, user and item
+    tmp_dir = str(tmpdir.mkdir("load_pandas_df"))

```

Comment:
some questions about this:
1) is tmpdir a pathlib object?
2) is `str(pathlib_path)` the recommended way to get the string from the path object?
3) do you actually need to pass a str or can you directly pass the object

Reply:
1: py.path.local path object
2: tested and we can just pass the object in this case (which will be converted as a string implicitly). If we have the case where implicit conversion doesn't work, then we should use str() that case.
3: answered in 2

Reply:
Ok, there is a way to get the string path in py.path, which is `strpath` property. But if pytest changes py.path to pathlib, it will break the test because pathlib doesn't have strpath. I think just explicitly calling `str()` is the easiest way to make it work on any cases. What do you think?

Reply:
ah... you know what. if pytest updates to use pathlib, anyway mkdir or other functions will not work lol per @anargyri's point about pytest is using py.path which is not recommended to the production  code, what will be the best way then?

Reply:
They already have a factory built on pathlib:
https://docs.pytest.org/en/latest/tmpdir.html
"The tmp_path_factory is a session-scoped fixture which can be used to create arbitrary temporary directories from any other fixture or test.

It is intended to replace tmpdir_factory, and returns pathlib.Path instances."
https://docs.pytest.org/en/latest/reference.html#tmp-path-factory-factory-api

Reply:
yeah, I think what @anargyri suggested is what @loomlike and I were looking for https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture

Reply:
@miguelgfierro @anargyri unfortunately, no it is not. I tested and still see all the **last three** temp folders and files in those three folders. They are not cleaned up after the session done. "Session-scoped" means just the temp-factory "object" is session-scoped, not the physical file-system folders. So, still our concerns remain the same -- the garbage files on the machine.

We can manually clean up, but to do that, should use `shutil.rmtree` again, because `pathlib.Path` doesn't have the function to delete non-empty folders.

So far, `with TemporaryDirectory() as tmp_dir:` works best for our purpose: remove the folders and files with context manager + can create many folders within the same test-function safely.